### PR TITLE
[RISCV] Add Syntacore SCR3 processor definition

### PR DESCRIPTION
**Pull request by asi-sc**
Syntacore SCR3 is a microcontroller-class processor core. Overview: https://syntacore.com/products/scr3
This PR introduces two CPUs:
  * 'syntacore-scr3-rv32' which is rv32imc
  * 'syntacore-scr3-rv64' which is rv64imac

### CODE REVIEW

**Review by anton-afanasyev — APPROVED** [review]
LGTM

File: clang/test/Driver/riscv-cpus.c [hunk 1]
```diff
@@ -358,3 +358,21 @@
 
 // RUN: not %clang --target=riscv32 -### -c %s 2>&1 -mcpu=generic-rv32 -march=rv64i | FileCheck -check-prefix=MISMATCH-ARCH %s
 // MISMATCH-ARCH: cpu 'generic-rv32' does not support rv64
+
+// RUN: %clang --target=riscv32 -### -c %s 2>&1 -mcpu=syntacore-scr3-rv32 | FileCheck -check-prefix=MCPU-SYNTACORE-SCR3-RV32 %s
+// MCPU-SYNTACORE-SCR3-RV32: "-target-cpu" "syntacore-scr3-rv32"
+// MCPU-SYNTACORE-SCR3-RV32: "-target-feature" "+m" "-target-feature" "+c"
```
Comment (wangpc-pp) [inline clang/test/Driver/riscv-cpus.c:364]: One `-target-feature` one line please.

File: clang/test/Driver/riscv-cpus.c [hunk 2]
```diff
@@ -358,3 +358,21 @@
...
+// MCPU-SYNTACORE-SCR3-RV32: "-target-feature" "+zicsr" "-target-feature" "+zifencei"
+// MCPU-SYNTACORE-SCR3-RV32: "-target-abi" "ilp32"
+
+// RUN: %clang --target=riscv32 -### -c %s 2>&1 -mtune=syntacore-scr3-rv32 | FileCheck -check-prefix=MTUNE-SYNTACORE-SCR3-RV32 %s
+// MTUNE-SYNTACORE-SCR3-RV32: "-tune-cpu" "syntacore-scr3-rv32"
```
Comment (wangpc-pp) [inline clang/test/Driver/riscv-cpus.c:369]: I think we don't need to test this, but I don't have a strong opinion here.

**Review by wangpc-pp — APPROVED** [review]
LGTM with nits.

File: clang/test/Driver/riscv-cpus.c (hunk 1 above)
Comment (asi-sc) [reply clang/test/Driver/riscv-cpus.c]: Addressed.

File: clang/test/Driver/riscv-cpus.c (hunk 2 above)
Comment (asi-sc) [reply clang/test/Driver/riscv-cpus.c:371]: This just copies the checks many processors do (rocket, veyron, xiangshan-nanhu, etc)

**Review by wangpc-pp — APPROVED** [review]

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Pull request merged**